test/software/amazon/ion/impl/IonWriterTestCase.java

```diff
@@ -496,16 +496,20 @@ public void testWriteValueCopiesFieldName()
         iw.writeValue(ir);
         iw.stepOut();
         assertEquals(data, reloadSingleValue());
+    }
 
+    @Test(expected = IllegalStateException.class)
+    public void testWriteValueCantCopyFieldName()
+            throws Exception
+    {
+        IonStruct data = struct("{a:{b:10}}");
         IonValue a = data.get("a");
-        ir = system().newReader(a);
+        IonReader ir = system().newReader(a);
         ir.next();
 
         iw = makeWriter();
         iw.stepIn(IonType.STRUCT);
-        iw.writeValue(ir);
-        iw.stepOut();
-        assertEquals(data, reloadSingleValue());
+        iw.writeValue(ir);  // should throw IllegalStateException ("Field name not set")
```

Comment:
We've been starting to shift to using JUnit's [`ExpectedException`](http://junit.org/junit4/javadoc/latest/org/junit/rules/ExpectedException.html) (instead of `@Test(expected=...)`) so that the particular line where the exception is expected may be isolated.